pkg/ingester/ingester.go

```diff
@@ -1554,7 +1614,9 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 		}
 
 		if activeSeries != nil && stats.succeededSamplesCount > oldSucceededSamplesCount {
-			activeSeries.UpdateSeries(nonCopiedLabels, ref, startAppend, numNativeHistogramBuckets)
+			idx, _ := i.getTSDB(userID).Head().Index()
```

Comment:
I think we can get the index only once per push, instead of one per series.

Reply:
addressed in commit https://github.com/grafana/mimir/pull/9733/commits/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-e1032332627c413a3010c66b54b22b6e9835cf152fa339e40cf0b11204f7241fR1431